### ISSUE TITLE
bipedhard doesn't converge

### ISSUE THREAD

**Issue by radsimu** [hardmaru/estool#11]
I tried both bipedhard and bipedhard_stoc. They jut don't converge. After like 150 generations they are still not showing any progress.
I am training with the following command:
    python train.py bipedhard -n 8 -t 4   
The best individuals are around -100 cumulative reward. Something must be wrong

**Comment by hardmaru** [comment]
Can you try with the -e 16 flag as well (as discussed in the blog post)?

In addition I would do -n 16 and -t 16 as well (or set the number so that
the product of the two options is 256)

**Comment by hardmaru** [comment]
Hi @radsimu 

Just checking in, did these options help the biped hard train in the end?

**Comment by radsimu** [comment]
python train.py bipedhard -n 16 -t 16 -e 16 is too slow on my i5 macbook.

I tried python train.py bipedhard -n 8 -t 4 -e 8, which should theoretically converge, but slower. Still, after 40 generations I see no sign of convergence, and the max is around -10 reward. Now that you mentioned it, I see the point of having -e bigger than 1 on a randomly generated environment, in general. But it should work with values bellow 16 as well. I will give it another go with 16 16 16

**Comment by radsimu** [comment]
there's a flag in model.py called "orig_mode" which is for a "hack for bipedhard's reward augmentation". Maybe that's the actual trick?

**Comment by hardmaru** [comment]
Hi @radsimu 

Yes, in this line:

orig_mode = True  # hack for bipedhard's reward augmentation during training (set to false for hack)

set `orig_mode = False` to remove the -100 score. I should have put this in the main documentation.

Thanks!

**Comment by ruiwang2uber** [comment]
Hello @hardmaru 

Thanks for the mention of reward augmentation. I found that without that, my agent stuck at the local minimum where it basically stopped in front of a stump.

To reproduce your result, should I always set orig_mode = False during training?

Thanks!

**Comment by hardmaru** [comment]
Yes, give that a try. I suggest trying it on the normal BipedalWalker first to make sure it works after a few hours (if it doesn't there might be other issues unrelated to this), and only then try it on BipedalWalkerHardcore. Let me know how it goes!

**Comment by ouyangzhuzhu** [comment]
hi~  friend, i trained the biped and the bipedhard, and i run the json file biped.openes.1.32.best.json
to see how the model works , and the it works very very slow that it takes 30 seconds to take a step..........i don't know why... is there  any settings i can use to make it work with nomal speed rather than such low...? @hardmaru can u help me friend?........

**Comment by ruiwang2uber** [comment]
@ouyangzhuzhu

I am kind of confused. 

What do you mean by "it takes 30 seconds to take a step"? 

Do you mean you trained the model for 30 seconds? 

Or the agent stands there for 30 seconds and then take a step?

**Comment by ouyangzhuzhu** [comment]
@ruiwang2uber 3ks friend your attention~, i trained the biped with "python train.py biped -o openes -n 8 -t 4" and after some time it best score is 331. Then I run the json file with "python model.py biped log/biped.openes.1.32.best.json". Then the window comes out but it works very slow, I tried again just now and the exact time is 4 seconds that is the screen refreshes very slowly - the refresh interval is up to four seconds.
It seems like when you watch a video , and u adjust the speed very slow , and the people in the video work very very slow....

**Comment by ruiwang2uber** [comment]
@ouyangzhuzhu 

Make sure you set "record_video = False"

BTW, which games you try: normal or hardcore version?

**Comment by MoMe36** [comment]
> Hi friend @ruiwang2uber :
> In model.py , it is already record_video = False .
> both the normal and hard that is biped and bipedhard have the same problem when i run their json files

There is a time.sleep(0.01) somewhere where the environment is rendered. My advice would be to comment that line.

**Comment by MoMe36** [comment]
> @ruiwang2uber 3ks friend your attention~, i trained the biped with "python train.py biped -o openes -n 8 -t 4" and after some time it best score is 331. Then I run the json file with "python model.py biped log/biped.openes.1.32.best.json". Then the window comes out but it works very slow, I tried again just now and the exact time is 4 seconds that is the screen refreshes very slowly - the refresh interval is up to four seconds.
> It seems like when you watch a video , and u adjust the speed very slow , and the people in the video work very very slow....

Could you share how long it took you to get the policy working with these parameters ? 
Thanks !
